src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1043,6 +1043,12 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
             if (!Context.DesignTimeMode)
             {
                 AcceptWhile(HtmlSymbolType.WhiteSpace);
+                if (Context.LastSpan.Kind == SpanKind.Transition)
+                {
+                    // Mark the whitespace at the end of a <text> tag as code.
+                    Span.CodeGenerator = new StatementCodeGenerator();
+                    Output(SpanKind.Code);
```

Comment:
@NTaylorMullen is your suggestion to add `Output(SpanKind.Markup)` (matching the `Output()` call at the end of this method) before accepting the non-newline whitespace?  If yes, :+1: from me.